app/workers/detect_invariants.rb

```diff
@@ -0,0 +1,27 @@
+# Detect state that *should* be impossible in the system and report them to Sentry
+class DetectInvariants
+  include Sidekiq::Worker
+
+  def perform
+    detect_unprocessed_feedbacks
+  end
+
+  def detect_unprocessed_feedbacks
+    # Application choices with completed feedback, but still awaiting references
+    choices_in_wrong_state = begin
+      ApplicationChoice.where(status: 'awaiting_references', application_form: ApplicationForm.includes(:references).select(&:references_complete?))
```

Comment:
This query feels a bit heavy but I can't improve on it

Reply:
Yeah, I'm pretty sure we can do this in one SQL query, but I've forgotten how to SQL in the late eighties.